libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/helpers/SwipeToRefreshHelper.java

```diff
@@ -18,19 +17,58 @@
-        init(context, swipeRefreshLayout, listener);
+        init(swipeRefreshLayout, listener, android.R.color.holo_blue_dark);
     }
 
-    public void init(Context context, CustomSwipeRefreshLayout swipeRefreshLayout, RefreshListener listener) {
+    /**
+     * Helps {@link org.wordpress.android.util.widgets.CustomSwipeRefreshLayout} by passing the
+     * {@link android.support.v4.widget.SwipeRefreshLayout}, {@link RefreshListener}, and color(s).
+     *
+     * @param swipeRefreshLayout    {@link CustomSwipeRefreshLayout} for refreshing the contents
+     *                              of a view via a vertical swipe gesture.
+     * @param listener              {@link RefreshListener} notified when a refresh is triggered
+     *                              via the swipe gesture.
+     * @param colorResIds           Comma-separated color resource integers used in the progress
+     *                              animation. The first color will also be the color of the bar
+     *                              that grows in response to a user swipe gesture.
+     */
+    public SwipeToRefreshHelper(CustomSwipeRefreshLayout swipeRefreshLayout, RefreshListener listener,
+                                @ColorRes int... colorResIds) {
+        init(swipeRefreshLayout, listener, colorResIds);
+    }
+
+    /**
+     * Initializes {@link org.wordpress.android.util.widgets.CustomSwipeRefreshLayout} by assigning
+     * {@link android.support.v4.widget.SwipeRefreshLayout}, {@link RefreshListener}, and color(s).
+     *
+     * @param swipeRefreshLayout    {@link CustomSwipeRefreshLayout} for refreshing the contents
+     *                              of a view via a vertical swipe gesture.
+     * @param listener              {@link RefreshListener} notified when a refresh is triggered
+     *                              via the swipe gesture.
+     * @param colorResIds           Comma-separated color resource integers used in the progress
+     *                              animation. The first color will also be the color of the bar
+     *                              that grows in response to a user swipe gesture.
+     */
+    public void init(CustomSwipeRefreshLayout swipeRefreshLayout, RefreshListener listener,
+                     @ColorRes int... colorResIds) {
         mRefreshListener = listener;
         mSwipeRefreshLayout = swipeRefreshLayout;
         mSwipeRefreshLayout.setOnRefreshListener(this);
-        final TypedArray styleAttrs = obtainStyledAttrsFromThemeAttr(context, R.attr.swipeToRefreshStyle,
-                R.styleable.RefreshIndicator);
-        int color = styleAttrs.getColor(R.styleable.RefreshIndicator_refreshIndicatorColor, context.getResources()
-                .getColor(android.R.color.holo_blue_dark));
-        mSwipeRefreshLayout.setColorSchemeColors(color, color, color, color);
+        mSwipeRefreshLayout.setColorSchemeResources(colorResIds);
```

Comment:
`obtainStyledAttrsFromThemeAttr()` is now never used. Can we remove it?

It also means that the color can no longer be set via the xml layout. Is that intentional?

Reply:
> `obtainStyledAttrsFromThemeAttr()` is now never used. Can we remove it?

Yep.  I removed that in d6906506c747d800005920afff6c06f1d06ae8d2.

> It also means that the color can no longer be set via the xml layout. Is that intentional?

I don't think that's an option with [SwipeRefreshLayout](https://developer.android.com/reference/android/support/v4/widget/SwipeRefreshLayout.html) either.  That along with the new helper helper method and the [new constructor](https://github.com/wordpress-mobile/WordPress-Android/pull/6644/files/bdc97c975f8d62ce426e03aeebe77ae6cf90f485#diff-1acd0c3685d5dcdcec9723ff6494909eR49) requiring at least one color resource assigned programmatically, I'm fine with removing the XML attribute.